File: problems/HackerRank/migratoryBirds.ts
function migratoryBirds(arr: number[]): number {
  let result = 1;

  let counter = [0, 0, 0, 0, 0] as number[];

  for (let ele of arr) {
    counter[ele - 1] += 1;
  }

  let max = counter[0];

  for (let i = 0; i < counter.length; i++) {
    if (counter[i] > max) {
      max = counter[i];
      result = i + 1;
    }
  }

  return result;
}

console.log(migratoryBirds([1, 1, 2, 2, 3]));

File: problems/HackerRank/week3/migratoryBirds.py
def migratoryBirds(arr):
    counter = [0] * 6
    ans = 0
    
    for ele in arr:
        counter[ele] += 1
        
    type_max = counter[1]
    for i in range(1, len(counter)):
        if counter[i] > type_max:
            type_max = counter[i]
            ans = i
    
    return ans